src/application-insights/azext_applicationinsights/util.py

```diff
@@ -70,3 +70,25 @@ def get_linked_properties(cli_ctx, app, resource_group, read_properties=None, wr
     else:
         linked_write_properties = [f'{tmpl}/{roles[write_properties]}']
     return linked_read_properties, linked_write_properties
+
+
+class Track1Credential:
```

Comment:
I suggest putting this class inside `src/application-insights/azext_applicationinsights/_client_factory.py` and make it private as `_Track1Credential`.